crates/iceberg/src/scan.rs

```diff
@@ -163,6 +178,54 @@ impl TableScan {
 
         Ok(iter(file_scan_tasks).boxed())
     }
+
+    /// Transforms a stream of FileScanTasks from plan_files into a stream of
+    /// Arrow RecordBatches.
+    pub fn open(&self, mut tasks: FileScanTaskStream) -> crate::Result<ArrowRecordBatchStream> {
```

Comment:
This method body's implementation is correct for one parquet file. For a stream of input files, we can use [futures::StreamExt::flat_map](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.flat_map) to combine them into a  stream of record batches.

Reply:
> Putting `open` in `TableScan` seems weird to me. The `TableScan` has two responsibilities: generate the task and open the task and it will make it complex. Can we separate it out to the type, e.g. xxReader.🤔

Sure, I can do that.